standards/application/messaging-api.md

```diff
@@ -0,0 +1,944 @@
+
+##### `error`
+An optional property that, if present, indicates that an error occurred during the processing of the message—whether while sending or upon receiving.
+This may include network, protocol, decoding errors, or other implementation-specific issues.
+Once set, the error persists; however, if a message that initially encountered an error (e.g., associated with a `RequestID`) is later successfully transmitted, the error information MUST be cleared.
+
+#### Methods
+
+##### `getByRequestId`
+Retrieves the `MessageRecord` corresponding to the specified `RequestID` associated with a send request for a particular message.
+If the provided `RequestID` is invalid or no associated `MessageRecord` exists,
+the method MUST return nothing.
+
+##### `getByHash`
+Retrieves the `MessageRecord` corresponding to the provided `message hash`.
+The hash is computed deterministically based on the message content,
+as described in the [MESSAGE](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/14/message.md#deterministic-message-hashing) specification.
+If the provided hash is invalid or no associated `MessageRecord` exists,
+the method MUST return nothing.
+
+##### `getByContentTopic`
+Retrieves an array of `MessageRecord` objects corresponding to messages that were circulated under the specified `contentTopic`.
+The method accepts two optional parameters for pagination:
+- `skip`: A non-negative integer specifying the number of initial matching records to bypass. If `skip` exceeds the total number of matching records, the method MUST return an empty array.
+- `take`: A non-negative integer specifying the maximum number of records to return. If `take` is 0, the method MUST return an empty array. If `take` is greater than the number of available records, all remaining matching records are returned.
+
+If both `skip` `and` take are omitted, the method MUST return all matching records.
+If no records match the specified `contentTopic`, an empty array MUST be returned.
+
+##### `getBySubscriptionId`
+Retrieves an array of `MessageRecord` objects corresponding to messages received under the specified `SubscriptionID` (as defined in the `Subscribe API` section).
+The method accepts two optional parameters for pagination:
+- `skip`: A non-negative integer specifying the number of initial matching records to bypass. If `skip` exceeds the total number of matching records, the method MUST return an empty array.
+- `take`: A non-negative integer specifying the maximum number of records to return. If `take` is 0, the method MUST return an empty array. If `take` is greater than the number of available records, all remaining matching records are returned.
+
+If both `skip` and `take` are omitted, the method MUST return all matching records.
+If no records match the specified `SubscriptionID`, an empty array MUST be returned.
+
```

Comment:
Too many APIs, especially with `getPendingRequests` already there.

I would say pick ONE way to check on message status at first, and we can then extend as we learn from dogfooding.